demos/segmentation_demo/main.cpp

```diff
@@ -89,178 +64,141 @@ int main(int argc, char *argv[]) {
-        for (auto & i : images) {
-            FormatReader::ReaderPtr reader(i.c_str());
-            if (reader.get() == nullptr) {
-                slog::warn << "Image " + i + " cannot be read!" << slog::endl;
-                continue;
-            }
-            /** Getting image data **/
-            std::shared_ptr<unsigned char> data(reader->getData(inputInfoItem.second->getTensorDesc().getDims()[3],
-                                                                inputInfoItem.second->getTensorDesc().getDims()[2]));
-            if (data.get() != nullptr) {
-                imagesData.push_back(data);
-            }
-        }
-        if (imagesData.empty()) throw std::logic_error("Valid input images were not found!");
-
-        /** Setting batch size using image count **/
-        network.setBatchSize(imagesData.size());
-        slog::info << "Batch size is " << std::to_string(network.getBatchSize()) << slog::endl;
 
-        inputInfoItem.second->setPrecision(Precision::U8);
-
-        // --------------------------- Prepare output blobs ----------------------------------------------------
-        slog::info << "Preparing output blobs" << slog::endl;
-
-        OutputsDataMap outputInfo(network.getOutputsInfo());
-        // BlobMap outputBlobs;
-        std::string firstOutputName;
-
-        for (auto & item : outputInfo) {
-            if (firstOutputName.empty()) {
-                firstOutputName = item.first;
-            }
-            DataPtr outputData = item.second;
-            if (!outputData) {
-                throw std::logic_error("output data pointer is not valid");
-            }
-
-            item.second->setPrecision(Precision::FP32);
+        ICNNNetwork::InputShapes inputShapes = network.getInputShapes();
+        if (inputShapes.size() != 1)
+            throw std::runtime_error("Demo supports topologies only with 1 input");
+        const std::string& inName = inputShapes.begin()->first;
+        SizeVector& inSizeVector = inputShapes.begin()->second;
+        if (inSizeVector.size() != 4 || inSizeVector[1] != 3)
+            throw std::runtime_error("3 channel 4 dimentional model's input is expected");
+        inSizeVector[0] = 1;  // set batch size to 1
+        network.reshape(inputShapes);
+
+        InputInfo& inputInfo = *network.getInputsInfo().begin()->second;
+        inputInfo.getPreProcess().setResizeAlgorithm(ResizeAlgorithm::RESIZE_BILINEAR);
+        inputInfo.setLayout(Layout::NHWC);
+        inputInfo.setPrecision(Precision::U8);
+
+        const OutputsDataMap& outputsDataMap = network.getOutputsInfo();
+        if (outputsDataMap.size() != 1) throw std::runtime_error("Demo supports topologies only with 1 output");
+        const std::string& outName = outputsDataMap.begin()->first;
+        Data& data = *outputsDataMap.begin()->second;
+        // alternative to resetting the output precision is to have a switch
+        // statement in postprocessing handling other precision types
```

Comment:
This comment doesn't quite make sense. If that alternative is better, why isn't it implemented instead? If _this_ alternative is better, then why have this comment at all?

Reply:
It was unexpected for me to get different types of data for deeplabv3 and any of 2 Intels' models. I think it can be useful to point it out.

As for picking a better solution, I can't say which is better. It is simpler to write/read just `data.setPrecision(Precision::FP32);`, but having switch can possibly bring faster processing.

Reply:
> I think it can be useful to point it out.

But you didn't point it out. This comment doesn't say why one might expect different data types.

> but having switch can possibly bring faster processing.

Can it, really? You're doing post-processing on the CPU, and x86 doesn't have half-precision floating-point arithmetic.

Reply:
> Can it, really? You're doing post-processing on the CPU, and x86 doesn't have half-precision floating-point arithmetic.

I think the default output type of deeplabv3 is sort of int representing class ids, so picking a solution with switch will eliminate casting from int to float by IE, and back from float to int by the demo.

> But you didn't point it out. This comment doesn't say why one might expect different data types.

I will consider extending the comment.

Reply:
Updated

Reply:
Alright, this is much better.